definitions/npm/react-router-dom_v6.x.x/flow_v0.104.x-/react-router-dom_v6.x.x.js

```diff
@@ -125,13 +125,66 @@ declare module "react-router-dom" {
     children?: React$Node
   |}>
 
-  declare export var Route: React$ComponentType<{|
-    caseSensitive?: boolean,
-    children?: React$Node,
-    element?: React$Element<any> | null,
-    index?: boolean,
-    path?: string,
-  |}>
+
+  declare type DataFunctionArgs = {|
+    request: Request;
+    params: Params<string>;
+    context?: any;
+  |}
+
+  declare export type LoaderFunction = (args: DataFunctionArgs) => Promise<Response> | Response | Promise<any> | any;
+
+  declare export type ActionFunction  = (args: DataFunctionArgs) => Promise<Response> | Response | Promise<any> | any;
+
+  declare export class DeferredData {
+    (data: mixed, responseInit?: ResponseOptions): void;
+    subscribe: mixed,
+    cancel: mixed,
+    resolveData: mixed,
+    done: mixed,
+    unwrappedData: mixed,
+    pendingKeys: mixed,
+  }
+
+  declare export type ShouldRevalidateFunction = (args: {|
+    currentUrl: URL;
+    currentParams: Params<string>;
+    nextUrl: URL;
+    nextParams: Params<string>;
+    formMethod?: "get" | "post" | "put" | "patch" | "delete";
+    formAction?: string;
+    formEncType?: "application/x-www-form-urlencoded" | "multipart/form-data";
+    formData?: FormData;
+    actionResult?: {|
+      type: 'data'|'deferred'|'redirect'|'error';
+      data?: any;
+      statusCode?: number;
+      headers?: Headers;
+      deferredData?: DeferredData;
+      status?: Number;
+      location?: string;
+      revalidate?: boolean;
+      error?: any;
+    |};
```

Comment:
the type of this prop is way more precise [on the other PR](https://github.com/flow-typed/flow-typed/pull/4418/files#diff-230ba295cc92e0fef6e9762eadf6cee929d0576a70f0198985c92295a12a418dR325)